services/waffle/waffle-label.service.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schema = Joi.array()
+  .items(
+    Joi.object({
+      label: Joi.object({
+        name: Joi.string().required(),
+        color: Joi.string()
+          .hex()
+          .required(),
+      }),
+      count: nonNegativeInteger,
+    })
+  )
+  .required()
+
+module.exports = class WaffleLabel extends BaseJsonService {
+  static get category() {
+    return 'issue-tracking'
+  }
+
+  static get route() {
+    return {
+      base: 'waffle/label',
+      pattern: ':user/:repo/:label?',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'waffle' }
+  }
+
+  static render({ label, color, count }) {
+    if (count === 'absent') {
+      return { message: count }
+    }
+    if (count === 0) {
+      color = '78bdf2'
+    }
+    return {
+      label,
+      message: metric(count),
+      color,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Waffle.io Label',
+        namedParams: {
+          user: 'ritwickdey',
+          repo: 'vscode-live-server',
+          label: 'closed',
+        },
+        staticPreview: this.render({
+          label: 'closed',
+          count: 2145,
+          color: 'A3D3F8',
+        }),
+      },
+    ]
+  }
+
+  async fetch({ user, repo }) {
+    const url = `https://api.waffle.io/${user}/${repo}/columns`
+    return this._requestJson({
+      schema,
+      url,
+      options: { qs: { with: 'count' } },
+      errorMessages: {
+        404: 'project not found',
+      },
+    })
+  }
+
+  transform({ json, label }) {
+    if (json.length === 0) {
+      return { count: 'absent' }
+    }
+
+    for (const col of json) {
+      if (col.label && col.label.name === label) {
+        return {
+          count: col.count,
+          color: col.label.color,
+        }
+      }
+    }
+
+    return { count: 0 }
+  }
+
+  async handle({ user, repo, label = 'ready' }) {
```

Comment:
I can see this is a default that we already had before refactoring, but I'm not sure this is a 'well-known' default as such. I don't think waffle boards necessarily have to have a 'ready' label. e.g: https://waffle.io/waffleio/waffle.io doesn't have one. If we're going to use this as a default for BC, can we document it in the example help.

Reply:
I've never used the service myself so I wasn't really sure what to do here either. I seriously considered removing that default (as it wasn't at all clear to me _why_ that was the default), but was more concerned about breaking existing users/badges that didn't explicitly specify a label. 

I can certainly add docs on the example stating that we will assume `ready` if no label is specified, but the only rationale I have for making that assumption is "because that's the way the legacy service implementation did it" 😄

Reply:
I suppose another option would be to make `label` a required route parameter in this service, and remove the default value of `ready`. I could add a redirector service that would catch the no label paths (`/waffle/label/user/repo`) and forward that along with `ready` to maintain backwards compatibility. 

We still get the same functional result, but at least the UI/net users will all see `label` as a required param

Reply:
Yeah that's a pretty nice solution. Good thinking :+1: